### DISCUSSION TITLE
QImg callback to fetch S3 signed image URL

### DISCUSSION THREAD

**Discussion by jblyberg** [quasarframework/quasar#15535] [Ideas / Proposals]
Hi all,

One of the challenges in hosting protected images on S3 is making sure the client has access to the file on S3. It would be super helpful if QImg offered a callback, similar to the factory prop in QUploader that could fetch a signed URL and provide it as the img src for the QImg component.

**Comment by pdanpdan** [comment]
What would be the difference from providing it the signed url from the start?

**Reply by jblyberg** [reply]
It would provide a lot of convenience. Instead of having separate code fetch and track signed URLs, a callback could take a resource Id and make an async callback, displaying spinner, waiting until images scroll onto the page (lazy loading), etc.
